17_differential_abundance.Rmd

```diff
@@ -10,6 +10,120 @@ chapterPreamble()
 A number of methods for differential abundance analysis are available,
 and reviewed elsewhere.
 
+## Tree-based methods
+
+### Group-wise associations testing based on balances
+
+[TreeSummarizedExperiment](https://bioconductor.org/packages/release/bioc/html/TreeSummarizedExperiment.html) 
+frequently includes a Phylogenetic tree along with associated data about the experiment (at `colData`), that holds covariates which can be used for analyzing group-wise associations. 
+
+Such an analysis could be performed with the function `pibble` from the `fido` package, that offers a Multinomial Logistic-Normal Linear Regression model; see [vignette](https://jsilve24.github.io/fido/articles/introduction-to-fido.html) of package.
+
+The following presents such an exemplary analysis based on the [Sprockett et al. (2020)](https://doi.org/10.1038/s41467-020-17541-6) available through `microbiomeDataSets` package.
+
+```{r, message=FALSE, warning=FALSE}
+if (!require(fido)){
+  # installing the fido package
+  devtools::install_github("jsilve24/fido")
+}
+```
+
+Loading the libraries and importing data:
+
+```{r, message=FALSE, warning=FALSE}
+library(fido)
+library(mia)
+library(microbiomeDataSets)
+
+tse <- SprockettTHData()
+names(colData(tse))
+```
+
+We pick three covariates ("Sex","Age_Years","Delivery_Mode") during this analysis as an example, and beforehand we check for missing data:
+
+```{r}
+cov_names <- c("Sex","Age_Years","Delivery_Mode")
+na_counts <- apply(is.na(colData(tse)[,cov_names]), 2, sum)
+na_summary<-as.data.frame(na_counts,row.names=cov_names)
+na_summary
+```
+
+We drop samples with na values at the covariates (features) under analysis: 
+
+```{r}
+tse <- subsetSamples(tse, !is.na(colData(tse)$Delivery_Mode))
+tse <- subsetSamples(tse, !is.na(colData(tse)$Age_Years))
+tse
+```
```

Comment:
This is ok, but `subsetSamples` are not really encouraged to use. https://microbiome.github.io/mia/reference/subsetSamples.html

I think these would go like this
```
tse <- tse[ , !is.na(colData(tse)$Delivery_Mode) ]
tse <- tse[ , !is.na(colData(tse)$Age_Years) ]
tse
```